src/main/java/dev/blachut/svelte/lang/parsing/js/SvelteJSParser.kt

```diff
@@ -16,6 +21,40 @@ class SvelteJSParser(builder: PsiBuilder) : ES6Parser<ES6ExpressionParser<*>, ES
                 // TODO Try to crate lazy element that splits variable and $ prefix
                 return super.getVariableElementType()
             }
+
+            override fun parseDialectSpecificSourceElements(marker: PsiBuilder.Marker): Boolean {
+                return tryParseReactiveDeclaration(marker) || super.parseDialectSpecificSourceElements(marker)
+            }
+
+            private fun tryParseReactiveDeclaration(marker: PsiBuilder.Marker): Boolean {
+                val dollarIdentifier = builder.tokenType == JSTokenTypes.IDENTIFIER && builder.tokenText == "$"
+                val reactiveLabel = dollarIdentifier && builder.lookAhead(1) === JSTokenTypes.COLON
+                // Treat reactive statements as normal labelled statements
+                val reactiveDeclaration = reactiveLabel && isIdentifierToken(builder.lookAhead(2))
+                if (!reactiveDeclaration) return false
+
+                if (isAfterModifierKeyword()) {
+                    builder.error("reactive declarations can't be exported")
+                }
+
+                builder.advanceLexer()
+                builder.advanceLexer()
```

Comment:
Why calling `builder.advanceLexer()` twice?

Reply:
because `$` and `:` are two separate tokens